lib/DXIL/DxilPDB.cpp

```diff
@@ -186,15 +176,29 @@ struct MSFWriter {
   }
 
   void WriteToStream(raw_ostream &OS) {
-    MSF_SuperBlock SB = CalculateSuperblock();
-    const uint32_t NumDirectoryBlocks = GetNumBlocks(SB.NumDirectoryBytes);
-    const uint32_t StreamDirectoryAddr = SB.BlockMapAddr;
-    const uint32_t BlockAddrSize =
-        NumDirectoryBlocks * sizeof(support::ulittle32_t);
-    const uint32_t NumBlockAddrBlocks = GetNumBlocks(BlockAddrSize);
-    const uint32_t StreamDirectoryStart =
-        StreamDirectoryAddr + NumBlockAddrBlocks;
-    const uint32_t StreamStart = StreamDirectoryStart + NumDirectoryBlocks;
+    const uint32_t StreamDirectorySizeInBytes = CalculateStreamDirectorySize();
+    const uint32_t StreamDirectoryNumBlocks =
+        GetNumBlocks(StreamDirectorySizeInBytes);
+
+    const uint32_t BlockAddrSizeInBytes =
+        StreamDirectoryNumBlocks * sizeof(support::ulittle32_t);
+    const uint32_t BlockAddrNumBlocks = GetNumBlocks(BlockAddrSizeInBytes);
+
+    const uint32_t BlockAddrStart = 3;
+    const uint32_t StreamDirectoryStart = BlockAddrStart + BlockAddrNumBlocks;
+    const uint32_t StreamStart =
+        StreamDirectoryStart + StreamDirectoryNumBlocks;
+
+    MSF_SuperBlock SB = {};
+    {
+      memcpy(SB.MagicBytes, kMsfMagic, sizeof(kMsfMagic));
+      SB.BlockSize = kMsfBlockSize;
+      SB.NumDirectoryBytes = StreamDirectorySizeInBytes;
+      SB.NumBlocks = 3 /*super block + FPM1 + FPM2*/ + m_NumStreamBlocks +
+                     StreamDirectoryNumBlocks + BlockAddrNumBlocks;
```

Comment:
Just because it was a little hard to glean amongst all the renaming and refactoring of value calculations. I think the only functional change here is that BlockAddrNumBlocks is now included in this calculation. That seems consistent with the description, but I wanted to confirm my thinking.

Reply:
Moved that magic number to a constant now.